### PR TITLE
Simplify the usage of dict

### PR DESCRIPTION
**Pull request by yqkcn**
### What problem does this PR solve?
Simplify the usage of dictionaries

### Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

### CODE REVIEW

**Comment by KevinHuSh** [comment]
Really impressive!

This alteration changes the out comes though.

**Comment by yqkcn** [comment]
got it...

**Comment by yqkcn** [comment]
> Really impressive!
> 
> This alteration changes the out comes though.

Please review again, I feel like I didn't misunderstand this time

File: graphrag/entity_resolution.py [hunk 1]
```diff
@@ -93,16 +93,12 @@ def __call__(self, graph: nx.Graph, prompt_variables: dict[str, Any] | None = No
...
-                    if a == b:
-                        continue
-                    if self.is_similarity(a, b) and (b, a) not in candidate_resolution_tmp:
-                        candidate_resolution_tmp.append((a, b))
-            if candidate_resolution_tmp:
-                candidate_resolution[node_cluster[0]] = candidate_resolution_tmp
+            for a, b in itertools.permutations(v, 2):
+                if self.is_similarity(a, b) and (b, a) not in candidate_resolution_tmp:
+                    candidate_resolution_tmp.append((a, b))
+            candidate_resolution[k] = candidate_resolution_tmp or v
```
Comment (KevinHuSh) [inline graphrag/entity_resolution.py]:
```suggestion
            candidate_resolution[k] = candidate_resolution_tmp
```

**Pull request merged**